src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java

```diff
@@ -416,6 +424,16 @@ protected String getValues(List<JsonObject> entities) {
     return promise.future();
   }
 
+  private boolean isEncumbranceRestricted(Transaction transaction, Budget budget, Ledger ledger) {
+    return transaction.getTransactionType() == Transaction.TransactionType.ENCUMBRANCE && ledger.getRestrictEncumbrance()
+        && budget.getAllowableEncumbrance() != null;
+  }
+
+  private boolean isPaymentRestricted(Transaction transaction, Budget budget, Ledger ledger) {
+    return transaction.getTransactionType() == Transaction.TransactionType.PAYMENT && ledger.getRestrictExpenditures()
+      && budget.getAllowableExpenditure() != null;
+  }
+
   /**
    * Calculates remaining amount for encumbrance
    * [remaining amount] = (allocated * allowableEncumbered) - (allocated - (unavailable + available)) - (encumbered + awaitingPayment)
```

Comment:
Could you update calculation of remaining amount for encumbrance according to discussion on [WIKI](https://wiki.folio.org/display/FOLIJET/Finance+Data+Model#FinanceDataModel-Encumbrance)

Reply:
added expenditures into calculations formula